### PR TITLE
fix project search intermittent

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* fixes a new intermittent
* adds some docs

### CODE REVIEW

File: docs/mochitests.md [hunk 1]
```diff
@@ -34,6 +40,42 @@ reflected in the new firefox directory.
 * `yarn mochi -- --jsdebugger` opens a browser toolbox
 * `yarn mochi browser_dbg-editor-highlight` runs just one test
 
+## Writing Tests
+
+Here are a few tips for writing mochitests:
+
+* There are lots of great helper methods in [head]
+* Try to write async user actions that involve a user action like clicking a button or typing a key press followed by a redux action to listen for. For example, the user step in action involvesthe user clicking the step in button followed by the "stepIn" action firing.
```
Comment (codehag) [inline docs/mochitests.md]: maybe this could have a code snippit?

File: src/test/mochitest/browser_dbg-search-project.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
 /* Any copyright is dedicated to the Public Domain.
  * http://creativecommons.org/publicdomain/zero/1.0/ */
 
-function toggleProjectSearch() {
+function openProjectSearch(dbg) {
```
Comment (codehag) [inline src/test/mochitest/browser_dbg-search-project.js]: :+1:

File: src/test/mochitest/browser_dbg-search-project.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
   synthesizeKeyShortcut("CmdOrCtrl+Shift+f");
+  return waitForState(
+    dbg,
+    state => dbg.selectors.getActiveSearchState(state) === "project"
+  );
+}
+
+function closeProjectSearch(dbg) {
```
Comment (codehag) [inline src/test/mochitest/browser_dbg-search-project.js]: :+1:

**Comment by codehag** [comment]
some linting errors:

```
yarn lint-md

yarn lint-md v0.24.5
$ remark -qf *.md src configs docs 
docs/mochitests.md
     6:5-6:36  warning  Link to unknown heading: `writing-Tests`. Did you mean `writing-tests`  remark-validate-links  remark-validate-links
  47:45-47:51  warning  Link to unknown heading in `src/test/mochitest/head.js`: `L206`         remark-validate-links  remark-validate-links

⚠ 2 warnings
error Command failed with exit code 1.

yarn lint-md returned exit code 1
```

File: docs/mochitests.md (hunk 1 above)
Comment (jasonLaster) [inline docs/mochitests.md]: yeah - we should revisit this section the next time we come in here. I think it could be better factored :)

**Pull request merged**